### PR TITLE
Add Semptian card support, capi_reset.sh and some refinements.

### PR DESCRIPTION
**Pull request by luyong6**
Signed by Lu Yong (luyong@cn.ibm.com)

### CODE REVIEW

File: capi_reset.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+
+if [ -n "$1" ]; then
+	echo user > /sys/class/cxl/card$1/load_image_on_perst
```
Comment (kenhill) [inline capi_reset.sh]: I would prefer using the `factory` partition for this.

File: psl-devices [hunk 2]
```diff
@@ -7,8 +7,8 @@
 0x3003 AlphaDataKU60 Xilinx
 0x0604 AlphaData7v3 Xilinx
 0x0605 AlphaDataKU60 Xilinx
-0x0606 AlphaData Xilinx
-0x0607 AlphaData Xilinx
+0x0606 SemtpianNSA121A Xilinx
```
Comment (kenhill) [inline psl-devices]: These card types do not match the information I have for PSL External Releases. Are these subIDs coming from IBM?

File: src/capi_flash_semptian_user.c [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+  vendor = temp & 0xFFFF;
+  device = (temp >> 16) & 0xFFFF;  
+  printf("Device ID: %04X\n", device);
+  printf("Vendor ID: %04X\n", vendor);
+
+  lseek(CFG, 44, SEEK_SET);
+  CHECK( read(CFG, &temp, 4) );
+  subsys = (temp >> 16) & 0xFFFF;
+
+  if ( (vendor != 0x1014) || ( device != 0x0477) || (subsys != 0x0607)) {
```
Comment (kenhill) [inline src/capi_flash_semptian_user.c]: Does this code work for 0x0606 as well?

**Comment by kenhill** [comment]
Opened  luyong6/capi-utils#1
I will merge this request if you approve of my changes

**Comment by luyong6** [comment]
Hi Ken, 
I merged your changes, and add a new commit de4d6d5
1) You changed the install point to usr/local/lib, but didn't update all of the occurances in capi-flash-utils.sh. I also update the description in README.md
2) add 0x0606 into src, although currently I think no one uses it. 

I have tested "make install" "capi-reset" and "capi-flash-script" in my system. They worked now.

File: psl-devices (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsvogt) [reply psl-devices]: There is no matching flasher C code for SemptianNSA121**A** - even after the typo fix

File: src/capi_flash_semptian_user.c [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+      cp = 0;
+      ec = 1;
+    }
+  }
+  printf("\n");
+
+  off_t fsize;
+  struct stat tempstat;
+  int num_blocks;
+  address = 0;  //the user partion.
```
Comment (jsvogt) [inline src/capi_flash_semptian_user.c]: Does Semptian have the user partition at address 0? Is there a factory partition at all? If yes, how does it make sure the images switch over? The image build then also requires matching parameters such as BITSTREAM.CONFIG.TIMER_CFG

File: psl-devices (hunk 2 above)
[the remark this answers is not in the document]
Comment (luyong6) [reply psl-devices]: Well, SemptianNSA121A is not enabled for SNAP. but it has the card with CAPI. So capi-utils can still keep the information. I will add "building executable binary" it in Makefile.

File: src/capi_flash_semptian_user.c (hunk 4 above)
Comment (kenhill) [reply src/capi_flash_semptian_user.c]: @jsvogt, the Alpha-Data KU115 and Bittware boards also use 0x0 for the user partition.

**Pull request merged**